### PR TITLE
Add correct line number to OpDebugFunction and OpDebugScope for function

### PR DESCRIPTION
**Pull request by chaocNV**
1. Pull OpDebugFunction, OpDebugScope and OpDebugVariable for params out of makeFunctionEntry.
2. Put above in a separate function called setupDebugFunctionEntry, which also accept line number and set it correctly in builder.
3. Call setupDebugFunctionEntry in makeFunction. Also special case handle entry function since it's created ealier elsewhere.

### CODE REVIEW

**Comment by chaocNV** [comment]
@arcady-lunarg @ncesario-lunarg Could you help review and if looks good, merge it?

**Review by ncesario-lunarg — COMMENTED** [review]
I thinks looks good on the surface, but it would be good to get @jeremy-lunarg's input as I believe he was the original implementer of this (I think?).

**Review by jeremy-lunarg — APPROVED** [review]
A couple of spelling corrections but LGTM. Thanks!

File: SPIRV/SpvBuilder.cpp [hunk 1]
```diff
@@ -2163,13 +2180,13 @@ Id Builder::makeDebugFunction([[maybe_unused]] Function* function, Id nameId, Id
     type->addImmediateOperand(NonSemanticShaderDebugInfo100DebugFunction);
     type->addIdOperand(nameId);
     type->addIdOperand(debugId[funcTypeId]);
-    type->addIdOperand(makeDebugSource(currentFileId)); // Will be fixed later when true filename available
-    type->addIdOperand(makeUintConstant(currentLine)); // Will be fixed later when true line available
+    type->addIdOperand(makeDebugSource(currentFileId)); // TODO: This points to file of definiation instead of declaration
```
[the remark this answers is not in the document]
Comment (chaocNV) [reply SPIRV/SpvBuilder.cpp]: thanks, fixed.

**Comment by chaocNV** [comment]
> @ncesario-lunarg

All review comment resolved. Could you help merge if it looks good to you?

**Comment by ncesario-lunarg** [comment]
@chaocNV if you are able to run the tests, can you please take a look at this failing test: https://github.com/KhronosGroup/glslang/actions/runs/6623976670/job/17992038094?pr=3357#step:11:3400

It looks like the test result file just needs to be updated.

**Comment by chaocNV** [comment]
> @chaocNV if you are able to run the tests, can you please take a look at this failing test: https://github.com/KhronosGroup/glslang/actions/runs/6623976670/job/17992038094?pr=3357#step:11:3400
> 
> It looks like the test result file just needs to be updated.

All fixed. Thanks for pointing out

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**